### ISSUE TITLE
cannot parse unicode character 

### ISSUE THREAD

**Issue by sonwh98** [michel-kraemer/citeproc-java#17]
org.jbibtex.TokenMgrError: Lexical error at line 1, column 35. Encountered: "\u00e7" (231), after : "" at org.jbibtex.LaTeXParserTokenManager.getNextToken(LaTeXParserTokenManager.java:303) at org.jbibtex.LaTeXParser.jj_scan_token(LaTeXParser.java:408) at org.jbibtex.LaTeXParser.jj_3R_6(LaTeXParser.java:270) at org.jbibtex.LaTeXParser.jj_3_3(LaTeXParser.java:253) at org.jbibtex.LaTeXParser.jj_2_3(LaTeXParser.java:242) at org.jbibtex.LaTeXParser.String(LaTeXParser.java:190) at org.jbibtex.LaTeXParser.Object(LaTeXParser.java:66) at org.jbibtex.LaTeXParser.ObjectList(LaTeXParser.java:46) at org.jbibtex.LaTeXParser.LaTeX(LaTeXParser.java:21) at org.jbibtex.LaTeXParser.parse(LaTeXParser.java:16) at de.undercouch.citeproc.bibtex.BibTeXConverter.toItemData(BibTeXConverter.java:161) at de.undercouch.citeproc.bibtex.BibTeXConverter.toItemData(BibTeXConverter.java:143) at de.undercouch.citeproc.bibtex.BibTeXItemDataProvider.addDatabase(BibTeXItemDataProvider.java:32) at 

The author name has unicode  "\u00e7" which is causing this exception. Here is the citation being parsed
"@article{Bloem_2014,doi = {10.4204/eptcs.157.7},url = {http://dx.doi.org/10.4204/eptcs.157.7},year = 2014,month = {jul},publisher = {Open Publishing Association},volume = {157},pages = {34--50},author = {Roderick Bloem and RÃ¼diger Ehlers and Swen Jacobs and Robert KÃ¶nighofer},title = {How to Handle Assumptions in Synthesis},journal = {Electronic Proceedings in Theoretical Computer Science}}"

**Comment by Fico1021** [comment]
+1

**Comment by michel-kraemer** [comment]
I think you should report this issue to the author of jbibtex:
https://github.com/jbibtex/jbibtex/issues

It's not really related to citeproc-java. Of course, as soon as the issue has been fixed in jbibtex I can update the library dependency.

Cheers,
Michel

**Comment by michel-kraemer** [comment]
Any news on this? Have you been able to resolve the issue?

**Comment by Fico1021** [comment]
What I have done:

I have a array with the common replacements:

private final String[] specialReplacements = {"à:\\`a","ô:\\^o","ê:\\^e","â:\\^a",
			"®:{\\textregistered}","ç:\\c{c}","ö:\\\"{o}","ä:\\\"{a}","ü:\\\"{u}",
			"Ö:\\\"{O}","Ä:\\\"{A}","Ü:\\\"{U}", "ï:\\\"{i}","Ï:\\\"{I}","ë:\\\"{e}", "Ë:\\\"{E}",
			"î:\\^i"};

I've made a hashmap with this array and then I developed this method:


public String scapeBibTex(String bibtex){
		
		 Iterator it = specialChars.entrySet().iterator();
		 
		 while (it.hasNext())
		 {
			 Map.Entry pairs = (Map.Entry)it.next();
			 bibtex = bibtex.replace((String) pairs.getKey(), (String) pairs.getValue());
		 }
			 
		
		return bibtex;
	}

Basically I've done the replacement by myself.

**Comment by michel-kraemer** [comment]
Thanks for letting us know! I think you should really open an issue in the jbibtex repository. I'm closing this one here now.
